src/query/storage/fanout/storage.go

```diff
@@ -128,6 +128,34 @@ func (s *fanoutStorage) FetchTags(ctx context.Context, query *storage.FetchQuery
 	return result, nil
 }
 
+func (s *fanoutStorage) CompleteTags(
+	ctx context.Context,
+	query *storage.CompleteTagsQuery,
+	options *storage.FetchOptions,
+) (*storage.CompleteTagsResult, error) {
+	var accumulatedTags storage.CompleteTagsResultBuilder
+
+	for _, store := range s.stores {
+		// TODO: once complete tags enabled locally,
+		// this should aggregate and merge results from all storages.
+		if store.Type() == storage.TypeRemoteDC {
```

Comment:
ugly :( Just use filters as done in read/write ?

Reply:
It uses a different query type to read/write

Reply:
I meant, maybe add a new filter for tags ?